conda-store-server/conda_store_server/utils.py

```diff
@@ -50,13 +50,52 @@ def chdir(directory: pathlib.Path):
         os.chdir(current_directory)
 
 
+def du(path):
+    """
+    Pure Python equivalent of du -sb
+
+    Based on https://stackoverflow.com/a/55648984/161801
+    """
+    if os.path.islink(path):
+        return os.lstat(path).st_size
+    if os.path.isfile(path):
+        st = os.lstat(path)
+        return st.st_size
+    apparent_total_bytes = 0
+    have = set()
+    for dirpath, dirnames, filenames in os.walk(path):
+        apparent_total_bytes += os.lstat(dirpath).st_size
+        for f in filenames:
+            fp = os.path.join(dirpath, f)
+            if os.path.islink(fp):
+                apparent_total_bytes += os.lstat(fp).st_size
+                continue
+            st = os.lstat(fp)
+            if st.st_ino in have:
+                continue
+            have.add(st.st_ino)
+            apparent_total_bytes += st.st_size
+        for d in dirnames:
+            dp = os.path.join(dirpath, d)
+            if os.path.islink(dp):
+                apparent_total_bytes += os.lstat(dp).st_size
+
+    return apparent_total_bytes
+
+
 def disk_usage(path: pathlib.Path):
     if sys.platform == "darwin":
         cmd = ["du", "-sAB1", str(path)]
-    else:
+    elif sys.platform == "linux":
         cmd = ["du", "-sb", str(path)]
+    else:
+        return str(du(path))
 
-    return subprocess.check_output(cmd, encoding="utf-8").split()[0]
+    output = subprocess.check_output(cmd, encoding="utf-8").split()[0]
+    if sys.platform == "darwin":
+        # mac du does not have the -b option to return bytes
+        output = str(int(output) * 512)
+    return output
```

Comment:
This "darwin" part needs to be moved to a different PR since it has nothing to do with Windows.

Reply:
Why?

Reply:
I mean, it can stay if you want. I just dislike having unrelated changes in PRs. This PR is called "Windows support", this part is for macOS. Here's why:

- If you want to keep it, this now requires me to test this part on macOS before we merge (both on x86_64 and ARM64).
- If there are problems with this part on macOS, this will block the PR from being merged even if the Windows part is fine.
- If problems are discovered later, we cannot just use `git-revert` to rollback and need to manually create a PR, which takes longer.

Reply:
The reason I did this here is because you requested it in your previous review https://github.com/conda-incubator/conda-store/pull/559#discussion_r1341985604. If you want to refactor it or split it out, that's fine. Based on the discussion the standup last week, my impression was that it isn't actually that important that this be super accurate.